zerver/models.py

```diff
@@ -284,18 +284,30 @@ def name_changes_disabled(realm):
         return settings.NAME_CHANGES_DISABLED
     return settings.NAME_CHANGES_DISABLED or realm.name_changes_disabled
 
+class DuplicateRealmAliasException(Exception):
+    pass
+
 class RealmAlias(models.Model):
     realm = models.ForeignKey(Realm, null=True) # type: Optional[Realm]
     # should always be stored lowercase
     domain = models.CharField(max_length=80, db_index=True) # type: Text
+    subdomains_allowed = models.BooleanField(default=False)
 
-def can_add_alias(domain):
-    # type: (Text) -> bool
-    if settings.REALMS_HAVE_SUBDOMAINS:
+    class Meta(object):
+        unique_together = ("realm", "domain", "subdomains_allowed")
+
+    def save(self, *args, **kwargs):
+        # type: (List[Any], Dict[Any, Any]) -> None
+        if not can_add_alias(self.domain, self.subdomains_allowed):
+            raise DuplicateRealmAliasException("A Realm for %s already exists." % (self.domain))
```

Comment:
-> "Another organization has already claimed the domain %(domain)s"
(we're using organization rather than realm for any text that can reach the user.)